staging/src/k8s.io/apiserver/pkg/server/options/encryptionconfig/config.go

```diff
@@ -426,3 +434,29 @@ func getEnvelopePrefixTransformer(config *apiserverconfig.KMSConfiguration, enve
 		Prefix:      []byte(prefix + config.Name + ":"),
 	}, nil
 }
+
+// parseEndpoint parses the endpoint to extract schema, host or path.
+func parseEndpoint(endpoint string) (string, error) {
```

Comment:
export this so NewGRPCService callers can parse endpoints. also, should this method go in this package or peer to NewGRPCService?